app/src/main/java/org/mozilla/focus/input/InputToolbarIntegration.kt

```diff
@@ -105,9 +123,61 @@ class InputToolbarIntegration(
     override fun start() {
         useCustomDomainProvider = settings.shouldAutocompleteFromCustomDomainList()
         useShippedDomainProvider = settings.shouldAutocompleteFromShippedDomainList()
+        if (fragment.components?.appStore?.state?.showStartBrowsingTabsCfr == true) {
+            observeStartBrowserCfrVisibility()
+        }
+    }
+
+    @VisibleForTesting
+    internal fun observeStartBrowserCfrVisibility() {
+        startBrowsingCfrScope = fragment.components?.appStore?.flowScoped { flow ->
+            flow.mapNotNull { state -> state.showStartBrowsingTabsCfr }
+                .ifChanged()
+                .collect { showStartBrowsingCfr ->
+                    if (showStartBrowsingCfr) {
+                        CFRPopup(
+                            anchor = toolbar.findViewById<AppCompatEditText>(R.id.mozac_browser_toolbar_background),
+                            properties = CFRPopupProperties(
+                                popupWidth = 256.dp,
+                                popupAlignment = CFRPopup.PopupAlignment.BODY_TO_ANCHOR_START,
+                                popupBodyColors = listOf(
+                                    ContextCompat.getColor(
+                                        fragment.requireContext(),
+                                        R.color.cfr_pop_up_shape_end_color,
+                                    ),
+                                    ContextCompat.getColor(
+                                        fragment.requireContext(),
+                                        R.color.cfr_pop_up_shape_start_color,
+                                    ),
+                                ),
+                                dismissButtonColor = ContextCompat.getColor(
+                                    fragment.requireContext(),
+                                    R.color.cardview_light_background,
+                                ),
+                                popupVerticalOffset = 0.dp,
+                            ),
+                            onDismiss = { onDismissStartBrowsingCfr() },
```

Comment:
The `onDismiss` callback informs of the user taped on the X button or not - maybe the CFR was dismissed by pressing outside of it.
Do we need to consider these separate scenarios?

Reply:
Done . Based on this bug https://github.com/mozilla-mobile/focus-android/issues/7653 cfrs should not be dismissed if the users taps anywhere on the screen.